packages/yarnpkg-core/sources/formatUtils.ts

```diff
@@ -307,22 +303,15 @@ export function addLogFilterSupport(report: Report, {configuration}: {configurat
       }
     }
 
-    if (logFilterByRegExp.length === 0)
-      return defaultLevel;
-
-    for (const [regExp, logLevel] of logFilterByRegExp)
-      if (regExp.test(text))
-        return logLevel ?? defaultLevel;
```

Comment:
No more filtering by text?

Reply:
I'm not sure if filtering by text is a good idea:

- It's brittle and therefore dangerous and not user friendly:
  - if one of our log lines changes the log filters need to be updated
  - if we add a new very important log line that _partially looks like_ an existing line that someone might have ignored, this could lead to unwanted behaviour.
    To give an example: in the tests I've ignored messages matching `/lists build scripts/` because of messages like

     ```
     YN0005: │ lorem-ipsum@npm:1.2.3 lists build scripts, but its build has been explicitly disabled through configuration.
     ```

     What if a message like "Couldn't install git dependency X because it lists build scripts but the build failed" is added later on?
- Yaml and escaping can be confusing.
  For instance, this object contains one key `foo\.foo` (it's invalid yaml because of that but that's not the point, the point is that depending on chosen syntax you might need to single/double escape dots and other characters with a meaning in regular expressions)

   ```yaml
   loremIpsum:
     foo\.foo: bar
     'foo\.foo': bar
     "foo\\.foo": bar
   ```

I'm convinced that filtering by message name only is a safer way to start providing this feature. It's easy to add text filtering later on if it's deemed necessary, but impossible to remove it without making a breaking change.

Reply:
The problem is that a very likely use case will be to silence a message from one particular package. For instance:

```yaml
logFilter:
  "lorem-ipsum@npm:1.2.3 lists build scripts, but its build has been explicitly disabled through configuration": discard
```

We don't really have any other way to allow silencing those without silencing *all* YN0005 messages, which would have high risks to let an actual problem come in unnoticed 🤔

Reply:
We could support passing in the _entire_ message instead of a regular expression. That leaves the downside where messages pop up again if they change on our end, but it is less dangerous and less confusing.

I just realised the previous regular expression solution wouldn't even work properly because of the colours in the messages. A solution where the entire message string would be added to configuration would also need to account for these `\e[...` escape sequences.

Reply:
I've modified it to support full message comparison instead, including a test to verify that it works with terminal colours enabled.

Reply:
Sounds good, thanks! I think I'll make one last change before merging: turn

```yaml
logFilter:
  YN0005: discard
  "lorem-ipsum@npm:1.2.3 lists build scripts, but its build has been explicitly disabled through configuration": discard
```

Into:

```yaml
logFilters:
  - name: YN0005
    strategy: discard
  - text: lorem-ipsum@npm:1.2.3 lists build scripts, but its build has been explicitly disabled through configuration
    strategy: discard
```

I think it'll help keep the design space open (for instance, perhaps with this we could consider adding the plugin name as an additional key separate from `name` once we have support for plugin messages).

Reply:
Sounds good, thanks! I think I'll make one last change before merging: turn

```yaml
logFilter:
  YN0005: discard
  "lorem-ipsum@npm:1.2.3 lists build scripts, but its build has been explicitly disabled through configuration": discard
```

Into:

```yaml
logFilters:
  - name: YN0005
    strategy: discard
  - text: lorem-ipsum@npm:1.2.3 lists build scripts, but its build has been explicitly disabled through configuration
    strategy: discard
```

I think it'll help keep the design space open (for instance, perhaps with this we could consider adding the plugin name as an additional key separate from `name` once we have support for plugin messages).